packages/workspace-utils/src/index.ts

```diff
@@ -200,3 +208,22 @@ export const workspaceFastGlobDefaultOptions = {
   extglob: false,
   braceExpansion: false,
 };
+
+/**
+ * Update the course instance usages for workspace usage.
+ *
+ * @param param.workspace_id The ID of the workspace.
+ * @param param.duration The usage duration (in milliseconds).
+ */
+export async function updateCourseInstanceUsagesForWorkspace({
```

Comment:
This has to go in this file because we need to call it from within this package.

Outside the scope of this PR, could we move `grader_host` and `workspace_host` into `app/prairielearn` so we can share things like `db-types.ts` and `models/` without needing to put the shared code in a package? Or should we move all of this support code to a package?

Reply:
I sorta-kinda like the fact that these apps are currently isolated from each other, I think it makes it easier to avoid making everything an even bigger ball of mud. I acknowledge that this does add more friction in sharing code between things.

What you have now is fine; as you said, we can discuss and reconsider more in the future.